tensorly/decomposition/CMTF_ALS

```diff
@@ -0,0 +1,165 @@
+import tensorly as tl
+import tensorly.tenalg
+import tensorly.random
+import numpy as np
+
+
+def rescale_to_unit_frobenius(matrix):
+    # rescale every column of matrix
+    return matrix / tl.norm(matrix, axis=0)
+
+
+def solve_ls(A, B):
+    # solve ||B-AX||^2 (AX = B -> X = A^{-1} @ B)
+    X = tl.pinv(A) @ B
+    return X
+
+
+def calculate_f(X, Y, A, B, C, V):
+    # calculate value of f in cmtf
+    return 1 / 2 * tl.norm(X - tl.kruskal_tensor.kruskal_to_tensor((None, [A, B, C]))) ** 2 + 1 / 2 * tl.norm(
+        Y - A @ V.T)
+
+
+def factor_match_score_3d(A, B, C, V, A_hat, B_hat, C_hat, V_hat):
+    """ Calculates factor match score (FMS) for comparing true and estimated matrices.
+
+    Assume 3rd order tensor X and matrix Y are coupled in the 1th mode with
+    X = [[A, B, C]] and Y = A @ V.T
+    The FMS shows how close the estimation of A, B, C, V is to the true matrices.
+    An FMS of 1 is a perfect match. The worst value is 0.
+
+    Formula for FMS according to Equation 4
+    @article{acar2011all,
+      title={All-at-once optimization for coupled matrix and tensor factorizations},
+      author={Acar, Evrim and Kolda, Tamara G and Dunlavy, Daniel M},
+      journal={arXiv preprint arXiv:1105.3422},
+      year={2011}
+    }
+
+    Parameters
+    ----------
+    A : tl.tensor
+        true 1st factor matrix of tensor X
+    B : tl.tensor
+        true 2nd factor matrix of tensor X
+    C : tl.tensor
+        true 3rd factor matrix of tensor X
+    V : tl.tensor
+        true matrix for generating Y
+    A_hat : tl.tensor
+        estimated 1st factor matrix of tensor X
+    B_hat : tl.tensor
+        estimated 2nd factor matrix of tensor X
+    C_hat : tl.tensor
+        estimated 3rd factor matrix of tensor X
+    V_hat : tl.tensor
+        estimated matrix for generating Y
+
+    Returns
+    -------
+    factor match score : float
+        value between 0 and 1 that shows how good the estimation is, 1 is perfect
+
+    """
+
+    R = A.shape[1]
+
+    # calculate weights
+    lambda_ = tl.zeros(R)
+    lambda_hat = tl.zeros(R)
+    alpha_ = tl.zeros(R)
+    alpha_hat = tl.zeros(R)
+    xi = tl.zeros(R)
+    xi_hat = tl.zeros(R)
```

Comment:
We could get directly kruskal_tensors and further apply kruskal_normalise to them to get the lambda?